### PR TITLE
Fix code coverage compilation and tests

### PR DESCRIPTION
**Pull request by Dentosal**
The compilation was failing on the latest nightly. Most imporantly, it caught one test that was incorrectly disabled due to incorrect `cfg`-gating. Found in preparation for code coverage setup. After this PR, the following command should work for getting codecov:

```bash
cargo +nightly-2024-06-05 llvm-cov --html --branch --all-features
```

## Checklist
- [x] Breaking changes are clearly marked as such in the PR description and changelog
- [x] New behavior is reflected in tests
- [x] [The specification](https://github.com/FuelLabs/fuel-specs/) matches the implemented behavior (link update PR if changes are needed)

### Before requesting review
- [x] I have reviewed the code myself
- [ ] I have created follow-up issues caused by this PR and linked them here

### CODE REVIEW

File: crates/fuel-core/src/database/state.rs [hunk 1]
```diff
@@ -231,29 +232,23 @@ mod tests {
...
-            let original_state = state_groups
+            let mut original_state = state_groups
                 .into_iter()
                 .flatten()
-                .sorted()
+                .map(|v| v.value)
                 .collect::<Vec<_>>();
 
+            states_in_db.sort_by(|a, b| a.as_ref().cmp(b.as_ref()));
+            original_state.sort_by(|a, b| a.as_ref().cmp(b.as_ref()));
```
Comment (xgreenx) [inline crates/fuel-core/src/database/state.rs:250]: Before we were comparing `ContractStateConfig`, now we only compare values. Why do we want to change it?

Comment (Dentosal) [reply crates/fuel-core/src/database/state.rs]: 751c407b0 Now back to comparing both keys and values

File: crates/fuel-core/src/database/state.rs [hunk 2]
```diff
@@ -231,29 +232,26 @@ mod tests {
...
-                .sorted()
+                .map(|v| (v.key, v.value))
                 .collect::<Vec<_>>();
 
+            states_in_db.sort_by(|(ak, av), (bk, bv)| {
+                (ak.as_ref(), av.as_ref()).cmp(&(bk.as_ref(), bv.as_ref()))
+            });
+            original_state.sort_by(|(ak, av), (bk, bv)| {
+                (ak.as_ref(), av.as_ref()).cmp(&(bk.as_ref(), bv.as_ref()))
+            });
```
Comment (xgreenx) [inline crates/fuel-core/src/database/state.rs:253]: The previous code was much simpler, why we don't want to compare `ContractStateConfig`?

File: tests/tests/poa.rs [hunk 3]
```diff
@@ -110,7 +110,7 @@ mod p2p {
     // after the first_producer stops, second_producer should start producing blocks
     #[tokio::test(flavor = "multi_thread")]
     async fn test_poa_multiple_producers() {
-        const SYNC_TIMEOUT: u64 = 30;
+        const SYNC_TIMEOUT: u64 = 120;
```
Comment (xgreenx) [inline tests/tests/poa.rs:113]: Why do we need to do this? The test should pass in 10 seconds, we set 30 just because of CI. Why 120?

File: crates/services/upgradable-executor/src/lib.rs [hunk 4]
```diff
@@ -3,6 +3,13 @@
 #![deny(unused_crate_dependencies)]
 #![deny(warnings)]
 
+#[cfg(test)]
+mod _ignore {
+    #![allow(unexpected_cfgs)] // for cfg(coverage)
+    #[cfg(coverage)]
+    use ntest as _; // Only used outside cdg(coverage)
```
Comment (xgreenx) [inline crates/services/upgradable-executor/src/lib.rs:10]: You can move it to `executor.rs`
image

File: Cargo.lock [hunk 5]
```diff
@@ -150,9 +150,9 @@ dependencies = [
 
 [[package]]
 name = "anstyle-query"
-version = "1.0.3"
+version = "1.1.0"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "a64c907d4e79225ac72e2a354c9ce84d50ebb4586dee56c82b3ee73004f537f5"
```
Comment (xgreenx) [inline Cargo.lock:155]: Do we need to bump version or we can leave it for a separate PR?

File: tests/tests/poa.rs (hunk 3 above)
Comment (Dentosal) [reply tests/tests/poa.rs:113]: It's too slow in branch coverage mode. I'll just disable it.

File: Cargo.lock (hunk 5 above)
Comment (Dentosal) [reply Cargo.lock:155]: Apparently not. I had issues with an old `ahash` version, but cargo clean and reinstall was enough.

File: .cargo/config.toml
Comment (Dentosal) [inline .cargo/config.toml:1]: The filename without `.toml` is deprecated

File: Cargo.lock (hunk 5 above)
Comment (Dentosal) [reply Cargo.lock]: Removed the changes in 1c6f607e79dd0c8dbbee8c8610bb949c0bd45b6b

File: crates/services/upgradable-executor/src/lib.rs (hunk 4 above)
Comment (Dentosal) [reply crates/services/upgradable-executor/src/lib.rs]: Sure, done in 1c6f607e79dd0c8dbbee8c8610bb949c0bd45b6b

File: tests/tests/poa.rs (hunk 3 above)
Comment (Dentosal) [reply tests/tests/poa.rs]: 1c6f607e79dd0c8dbbee8c8610bb949c0bd45b6b

File: crates/fuel-core/src/database/state.rs (hunk 2 above)
Comment (Dentosal) [reply crates/fuel-core/src/database/state.rs]: The old version didn't compile, so it was quite easy for it to be simple. I've tried to improve this a bit in 1c6f607e79dd0c8.

**Review by xgreenx — APPROVED** [review]
